### PR TITLE
ISPN-5698 & 	ISPN-5684

### PR DESCRIPTION
**Pull request by galderz**
https://issues.jboss.org/browse/ISPN-5698
https://issues.jboss.org/browse/ISPN-5684

### CODE REVIEW

File: core/src/main/java/org/infinispan/statetransfer/StateTransferInterceptor.java [hunk 1]
```diff
@@ -171,7 +171,7 @@ public Object visitGetAllCommand(InvocationContext ctx, GetAllCommand command) t
 
          // We increment the topology id so that updateTopologyIdAndWaitForTransactionData waits for the next topology.
          // Without this, we could retry the command too fast and we could get the OutdatedTopologyException again.
-         if (trace) log.tracef("Retrying command because of topology change, current topology is %d: %s", command);
+         if (trace) log.tracef("Retrying command because of topology change, current topology is %d: %s", currentTopologyId(), command);
```
Comment (danberindei) [inline core/src/main/java/org/infinispan/statetransfer/StateTransferInterceptor.java]: Could you also add the missing parameter from `visitReadOnlyManyCommand`?

File: client/hotrod-client/src/test/java/org/infinispan/client/hotrod/GetAllCompatDistTest.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+   public void testGetAllWithCompatibility() {
+      RemoteCache<String, String> cache = client(0).getCache();
+      HashMap<String, String> cachedValues = new HashMap<>();
+      for(int i=0; i<100; i++){
+         String key = String.format("key-%d", i);
+         String value = String.format("value-%d", i);
+         cache.put(key, value);
+         cachedValues.put(key, value);
+      }
+      System.out.println("Key's added");
```
Comment (danberindei) [inline client/hotrod-client/src/test/java/org/infinispan/client/hotrod/GetAllCompatDistTest.java]: Logging

File: client/hotrod-client/src/test/java/org/infinispan/client/hotrod/GetAllCompatDistTest.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+import org.infinispan.configuration.cache.CacheMode;
+import org.infinispan.configuration.cache.ConfigurationBuilder;
+import org.testng.annotations.Test;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.testng.AssertJUnit.assertEquals;
+
+@Test(testName = "client.hotrod.Ispn5684Test", groups = "functional")
```
Comment (danberindei) [inline client/hotrod-client/src/test/java/org/infinispan/client/hotrod/GetAllCompatDistTest.java]: Test name

**Comment by galderz** [comment]
@danberindei Fixed all, also updated 7.2.x branch accordingly.

**Comment by tristantarrant** [comment]
Pulling

**Comment by tristantarrant** [comment]
@galderz I'm getting a bunch of OOMEs here
Failed tests: 
  PingOnStartupTest.testGetCacheWithPingOnStartupDisabledMultipleNodes:101 » OutOfMemory
  PingOnStartupTest.testGetCacheWorksIfNodeDown » OutOfMemory Java heap space
  PingOnStartupTest.testGetCacheWorksIfNodeNotDown » OutOfMemory Java heap space
  PingOnStartupTest.testTopologyFetched » OutOfMemory Java heap space
  DistFailOverRemoteIteratorTest>BaseIterationFailOverTest.testFailOver:38 » OutOfMemory
org.infinispan.client.hotrod.impl.iteration.MultiServerDistRemoteIteratorTest.clear(org.infinispan.client.hotrod.impl.iteration.MultiServerDistRemoteIteratorTest)
  Run 1: MultiServerDistRemoteIteratorTest>BaseMultiServerRemoteIteratorTest.clear:37->BaseMultiServerRemoteIteratorTest.lambda$clear$1:37 » OutOfMemory
  Run 2: PASS
  Run 3: PASS
  Run 4: PASS
  Run 5: PASS

org.infinispan.client.hotrod.marshall.EmbeddedCompatTest.createBeforeMethod(org.infinispan.client.hotrod.marshall.EmbeddedCompatTest)
  Run 1: EmbeddedCompatTest>SingleCacheManagerTest.createBeforeMethod:55->SingleCacheManagerTest.setup:31->createCacheManager:86 » OutOfMemory
  Run 2: PASS
  Run 3: PASS
  Run 4: PASS
  Run 5: PASS

**Comment by galderz** [comment]
@tristantarrant That's very strange...

**Comment by galderz** [comment]
@tristantarrant CI looks fine

**Comment by gustavonalle** [comment]
The OOME has been happening to me as well since a few of weeks, but not on every machine I run the build

**Comment by tristantarrant** [comment]
Pushed

**Pull request closed without merging**